### PR TITLE
[gen] bug in check_fetch causing unwanted output

### PR DESCRIPTION
**Pull request by hugookeeffe**
This PR solves a bug where a print statement in check fetch is triggered when using **diy7**, generating unwanted output.

### CODE REVIEW

File: gen/cycle.ml
Comment (relokin) [inline gen/cycle.ml:723]: Would this be useful if it was gated by a debug flag?

Comment (hugookeeffe) [reply gen/cycle.ml:723]: Hi @relokin thanks for comment! I'm not sure that a debug flag is necessary as using Warn.user_error will give the same result (as it will only be printed if debug mode is on, or if using diyone7, in which case it should indeed fail).

Comment (relokin) [reply gen/cycle.ml:723]: ah I sorry I spoke too fast and didn't realise that you added the call to str_node in the user_error.

**Review by relokin — APPROVED** [review]
LGTM. Planning to merge this tomorrow, unless I hear any objections :)

**Comment by maranget** [comment]
Fine by me. Thanks @hugookeeffe and @relokin. FYI. master has just been updated. Would you please rebase on master before merging?

**Comment by maranget** [comment]
Merged, thanks @hugookeeffe and @relokin.

**Pull request merged**
